octodns/provider/cloudflare.py

```diff
@@ -14,14 +14,14 @@
 from .base import BaseProvider
 
 
-class CloudflareAuthenticationError(Exception):
+class CloudflareError(Exception):
 
     def __init__(self, data):
         try:
             message = data['errors'][0]['message']
         except (IndexError, KeyError):
             message = 'Authentication error'
-        super(CloudflareAuthenticationError, self).__init__(message)
+        super(CloudflareError, self).__init__(message)
```

Comment:
Since removing it would be an API regressions I think I'd prefer to have `CloudflareAuthenticationError ` be renamed `CloudflareError` (as you've done) and then have a child class `CloudflareAuthenticationError(CloudflareError)`. You could then just treat 403 and 400 distinctly down below.